### PR TITLE
Hdadma : DMA flow unification 

### PR DESCRIPTION
**Pull request by awladzin**
It is a bit big package, but after consulting our DL Marcin, we decided that only in this way we will  not crash CI.

### CODE REVIEW

File: src/drivers/hda-dma.c [hunk 1]
```diff
@@ -117,43 +117,71 @@ static inline void hda_update_bits(struct dma *dma, uint32_t chan,
 {
 	io_reg_update_bits(dma_chan_base(dma, chan) + reg,  mask, value);
 }
+static uint64_t hda_work(void *data, uint64_t delay)
+{
+	struct dai *dai = (struct dai *)data;
+	struct dma_pdata *p = dma_get_drvdata(dma);
+		int32_t gval;
```
Comment (lgirdwood) [inline src/drivers/hda-dma.c]: indentation looks wrong.

File: src/drivers/hda-dma.c [hunk 2]
```diff
@@ -117,43 +117,71 @@ static inline void hda_update_bits(struct dma *dma, uint32_t chan,
...
-	 *
-	 * if (p->chan[channel].cb)
-	 *	p->chan[channel].cb(p->chan[channel].cb_data,
-	 *		DMA_IRQ_TYPE_LLIST,
-	 *		NULL);
-	 *
-	 */
+	*/
+
+	uint32_t dgbwp = host_dma_reg_read(dma, channel, DGBWP);
```
Comment (lgirdwood) [inline src/drivers/hda-dma.c]: All variable declerations should be at the top.

**Comment by lgirdwood** [comment]
oh, and please also delete the merge commit. thanks.

**Comment by plbossart** [comment]
Kind reminder that we try to use the Linux style, and I get tons of issues reported by checkpatch.pl.

CHECK: Comparison to NULL could be written "!config->elem_array.elems"
#53: FILE: src/audio/dai.c:269:
+	if (config->elem_array.elems == NULL) {

CHECK: Alignment should match open parenthesis
#55: FILE: src/audio/dai.c:271:
+		err = dma_sg_alloc(&config->elem_array,
+			config->direction,

CHECK: Comparison to NULL could be written "!config->elem_array.elems"
#108: FILE: src/audio/dai.c:317:
+	if (config->elem_array.elems == NULL) {

CHECK: Alignment should match open parenthesis
#110: FILE: src/audio/dai.c:319:
+		err = dma_sg_alloc(&config->elem_array,
+			config->direction,

CHECK: Comparison to NULL could be written "!dd->config.elem_array.elems"
#138: FILE: src/audio/dai.c:408:
+	if (dd->config.elem_array.elems == NULL) {

CHECK: Unnecessary parentheses around 'hc->current'
#218: FILE: src/audio/host.c:103:
+	if (++(hc->current) == hc->elem_array.count)

CHECK: if this code is redundant consider removing it
#289: FILE: src/audio/host.c:255:
+#if 0

CHECK: if this code is redundant consider removing it
#296: FILE: src/audio/host.c:262:
+#if 0

CHECK: if this code is redundant consider removing it
#338: FILE: src/audio/host.c:304:
+#if 0

CHECK: Lines should not end with a '('
#401: FILE: src/audio/host.c:380:
+	err = dma_sg_alloc(

CHECK: if this code is redundant consider removing it
#417: FILE: src/audio/host.c:396:
+#if 0

CHECK: Alignment should match open parenthesis
#918: FILE: src/include/sof/dma.h:276:
+int dma_sg_alloc(struct dma_sg_elem_array *elem_array, uint32_t direction,
+	uint32_t period_count, uint32_t period_bytes,

CHECK: Alignment should match open parenthesis
#970: FILE: src/ipc/dma-copy.c:91:
+int dma_copy_to_host(struct dma_copy *dc, struct dma_sg_config *host_sg,
+	int32_t host_offset, void *local_ptr, int32_t size)

CHECK: Alignment should match open parenthesis
#1003: FILE: src/ipc/dma-copy.c:256:
+int dma_copy_from_host(struct dma_copy *dc, struct dma_sg_config *host_sg,
+	int32_t host_offset, void *local_ptr, int32_t size)

CHECK: Alignment should match open parenthesis
#1142: FILE: src/lib/dma-trace.c:220:
+	err = dma_sg_alloc(&config.elem_array, config.direction,
+			elem_num, elem_size, elem_addr, 0);

CHECK: Alignment should match open parenthesis
#1166: FILE: src/lib/dma.c:93:
+int dma_sg_alloc(struct dma_sg_elem_array *elem_array, uint32_t direction,
+	uint32_t period_count, uint32_t period_bytes,

CHECK: Alignment should match open parenthesis
#1172: FILE: src/lib/dma.c:99:
+	elem_array->elems = rzalloc(RZONE_RUNTIME, SOF_MEM_CAPS_RAM,
+		sizeof(struct dma_sg_elem) * period_count);

CHECK: Comparison to NULL could be written "!elem_array->elems"
#1173: FILE: src/lib/dma.c:100:
+	if (elem_array->elems == NULL)

ERROR: Missing Signed-off-by: line(s)

CHECK: Please use a blank line after function/struct/union/enum declarations
#16: FILE: src/drivers/hda-dma.c:120:
 }
+static uint64_t hda_work(void *data, uint64_t delay)

CHECK: Blank lines aren't necessary after an open brace '{'
#36: FILE: src/drivers/hda-dma.c:139:
 {
+

CHECK: Please don't use multiple blank lines
#37: FILE: src/drivers/hda-dma.c:140:
+
+

ERROR: space required after that ',' (ctx:VxV)
#60: FILE: src/drivers/hda-dma.c:156:
+	work_schedule_default(dma,HDA_LINK_1MS_US);
 	                         ^

WARNING: suspect code indent for conditional statements (8, 12)
#63: FILE: src/drivers/hda-dma.c:159:
+	if ((dgbwp << 8) == 0 && (dgbrp << 8) != 0) {
+	    if (p->chan[channel].cb) {

WARNING: Statements should start on a tabstop
#64: FILE: src/drivers/hda-dma.c:160:
+	    if (p->chan[channel].cb) {

CHECK: Please don't use multiple blank lines
#74: FILE: src/drivers/hda-dma.c:167:
 
+

CHECK: Blank lines aren't necessary before a close brace '}'
#90: FILE: src/drivers/hda-dma.c:183:
 
+	}

CHECK: Alignment should match open parenthesis
#84: FILE: src/drivers/hda-dma.c:250:
+	work_schedule_default(hda_dma_copy(dma, channel,
+			p->chan[channel].cb_data),

**Comment by mUfoq** [comment]
@plbossart I think you have to escape '#' char cos you are referencing a lot of random pull requestes

**Comment by plbossart** [comment]
@mUfoq i fixed the comments, please fix your code...

**Comment by jiawang6** [comment]
CI observed building failure.

Failed Platform: baytrail apollolake cannonlake

configure: error: in `/build/sof/src':
configure: error: C compiler cannot create executables

**Comment by lgirdwood** [comment]
@jiawang6 this PR will be is on a version of master that does not have configure.ac fix. It will be rebased soon and will be getting merged into -next (after squash/rebase).

**Pull request closed without merging**